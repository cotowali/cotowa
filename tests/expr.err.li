fn f(a int) int {
  return a
}

echo(x)

f("a") + "a"
let x = "a" + f("a")

y = 10
y = "10"
0 = 10

f("a") |> f("a")

let arr = [0]
arr['10']
arr[0][0]
x[0]
