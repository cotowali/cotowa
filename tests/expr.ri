6 / 2 + 3 * 3
1 + 1 - 1
6 / 2 % 2
(1 + 2) * (4 - 2)
let v = 10
-v
+v

fn zero() int {
  let v = 0

  // evaluated but not used. this is valid but no output
  0
  v
  add(1, 2)

  return v
}

fn add(a int, b int) int {
  return a + b
}

fn as_int(v any) int {
  return v as int
}

zero()
add(3, 4)
add(as_int("1"), 2)
