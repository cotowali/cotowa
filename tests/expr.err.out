expr.err.li 5,7: undefined variable `x`
  > x
    ^
expr.err.li 7,2: mismatched types: `int` (left) and `string` (right)
  > f("a") + "a"
    ^^^^^^^^^^^^
expr.err.li 7,4: mismatched types: `int` (expected) and `string` (found)
  > "a"
    ^^^
expr.err.li 8,10: mismatched types: `string` (left) and `int` (right)
  > "a" + f("a")
    ^^^^^^^^^^^^
expr.err.li 8,18: mismatched types: `int` (expected) and `string` (found)
  > "a"
    ^^^
expr.err.li 10,2: undefined variable `y`
  > y
    ^
expr.err.li 11,2: undefined variable `y`
  > y
    ^
expr.err.li 12,2: invalid left-hand side of assignment
  > 0
    ^
expr.err.li 14,4: mismatched types: `int` (expected) and `string` (found)
  > "a"
    ^^^
expr.err.li 14,14: mismatched types: `int` (expected) and `string` (found)
  > "a"
    ^^^
expr.err.li 15,7: expected function call, but found int
  > 0
    ^
expr.err.li 18,6: mismatched types: `int` (expected) and `string` (found)
  > '10'
    ^^^^
expr.err.li 19,2: `int` does not support indexing
  > arr[0][0]
    ^^^^^^^^^
expr.err.li 20,2: `int` does not support indexing
  > x[0]
    ^^^^
